Code Fixes

lakshyacis opened the pull request:
messagetypes init flake and pylint fixes

PeterSurda commented on src/messagetypes/message.py:
```diff
@@ -1,24 +1,30 @@
         else:
             self.subject = unicode(str(data["subject"]), 'utf-8', 'replace')
-        if type(data["body"]) is str:
+        if isinstance(data["subject"], str):
```
should be `body` not `subject`

PeterSurda commented on src/messagetypes/message.py:
```diff
@@ -1,24 +1,30 @@
     def encode(self, data):
-        super(Message, self).encode()
+        """Encode a message"""
+        super(Message, self).__init__()
```
is this even necessary?

lakshyacis replied:
I removed encode and use __init__ instide of that. Because pylint shows so many errors like, old-style-class, variable should be defined under init, no-init called. So, i think that the best alternative to that.

PeterSurda commented on src/messagetypes/vote.py:
```diff
@@ -1,23 +1,31 @@
     def encode(self, data):
-        super(Vote, self).encode()
+        """Encode a vote"""
+        super(Vote, self).__init__()
```
I'm not sure this is necessary.

g1itch commented:
What was it? Some sort of obfuscation?

lakshyacis commented:
@g1itch as discussed with peter, we got some issues in the messagetypes during sending functionality. So, I discussed with peter and i fixed the issues getting in messgetypes files. and Solved flake8 and pylint errors. In this Pr, I changed the def encode  to def __init__ because pylint highlights so many errors and warnings like no-init,  variable should be defined under init etc. And the best solution is to change encode to init. So change the encode to init. We even test the code and the code works fine.

PeterSurda commented:
@g1itch what's the problem?

g1itch commented:
You changed one pylint warning by another and immediately suppressed it: `# pylint: disable=too-few-public-methods`.

g1itch commented:
> @g1itch what's the problem?

Why rename `encode()` to `__init__()`? Readability is made worse.

lakshyacis commented:
@g1itch so what I should do at this moment I mean should I keep them or undo it as before please suggest?

g1itch commented:
Well, I tried to do it myself. And maybe your idea with `__init__()` makes some sense...
```patch
diff --git a/src/messagetypes/__init__.py b/src/messagetypes/__init__.py
index 06783eac..823fae38 100644
--- a/src/messagetypes/__init__.py
+++ b/src/messagetypes/__init__.py
@@ -1,14 +1,26 @@
+import os
+from abc import ABCMeta, abstractmethod
 from importlib import import_module
-from os import path, listdir
-from string import lower
 
-from debug import logger
 import messagetypes
 import paths
+from debug import logger
+
 
 class MsgBase(object):
-    def encode(self):
-        self.data = {"": lower(type(self).__name__)}
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        self.subject = self.body = None
+        self.data = {"": type(self).__name__.lower()}
+
+    @abstractmethod
+    def encode(self, data):
+        pass
+
+    @abstractmethod
+    def decode(self, data):
+        pass
 
 
 def constructObject(data):
@@ -18,7 +30,9 @@ def constructObject(data):
     try:
         classBase = getattr(getattr(messagetypes, data[""]), data[""].title())
     except (NameError, AttributeError):
-        logger.error("Don't know how to handle message type: \"%s\"", data[""], exc_info=True)
+        logger.error(
+            "Don't know how to handle message type: \"%s\"",
+            data[""], exc_info=True)
         return None
     try:
         returnObj = classBase()
@@ -32,18 +46,19 @@ def constructObject(data):
     else:
         return returnObj
 
-if paths.frozen is not None:
+
+if paths.frozen:
     import messagetypes.message
     import messagetypes.vote
 else:
-    for mod in listdir(path.dirname(__file__)):
+    for mod in os.listdir(os.path.dirname(__file__)):
         if mod == "__init__.py":
             continue
-        splitted = path.splitext(mod)
-        if splitted[1] != ".py":
+        name, ext = os.path.splitext(mod)
+        if ext != ".py":
             continue
         try:
-            import_module(".{}".format(splitted[0]), "messagetypes")
+            import_module(".{}".format(name), "messagetypes")
         except ImportError:
             logger.error("Error importing %s", mod, exc_info=True)
         else:
diff --git a/src/messagetypes/message.py b/src/messagetypes/message.py
index f52c6b35..412b4a38 100644
--- a/src/messagetypes/message.py
+++ b/src/messagetypes/message.py
@@ -3,22 +3,18 @@ from messagetypes import MsgBase
 
 
 class Message(MsgBase):
-    def __init__(self):
-        return
-
     def decode(self, data):
         # UTF-8 and variable type validator
-        if type(data["subject"]) is str:
+        if isinstance(data["subject"], str):
             self.subject = unicode(data["subject"], 'utf-8', 'replace')
         else:
             self.subject = unicode(str(data["subject"]), 'utf-8', 'replace')
-        if type(data["body"]) is str:
+        if isinstance(data["body"], str):
             self.body = unicode(data["body"], 'utf-8', 'replace')
         else:
             self.body = unicode(str(data["body"]), 'utf-8', 'replace')
 
     def encode(self, data):
-        super(Message, self).encode()
         try:
             self.data["subject"] = data["subject"]
             self.data["body"] = data["body"]

```

g1itch commented:
Ideally I would have refactored `messagetypes` together with `helper_msgcoding`

lakshyacis commented:
> Well, I tried to do it myself. And maybe your idea with `__init__()` makes some sense...
> 
> ```diff
> diff --git a/src/messagetypes/__init__.py b/src/messagetypes/__init__.py
> index 06783eac..823fae38 100644
> --- a/src/messagetypes/__init__.py
> +++ b/src/messagetypes/__init__.py
> @@ -1,14 +1,26 @@
> +import os
> +from abc import ABCMeta, abstractmethod
>  from importlib import import_module
> -from os import path, listdir
> -from string import lower
>  
> -from debug import logger
>  import messagetypes
>  import paths
> +from debug import logger
> +
>  
>  class MsgBase(object):
> -    def encode(self):
> -        self.data = {"": lower(type(self).__name__)}
> +    __metaclass__ = ABCMeta
> +
> +    def __init__(self):
> +        self.subject = self.body = None
> +        self.data = {"": type(self).__name__.lower()}
> +
> +    @abstractmethod
> +    def encode(self, data):
> +        pass
> +
> +    @abstractmethod
> +    def decode(self, data):
> +        pass
>  
>  
>  def constructObject(data):
> @@ -18,7 +30,9 @@ def constructObject(data):
>      try:
>          classBase = getattr(getattr(messagetypes, data[""]), data[""].title())
>      except (NameError, AttributeError):
> -        logger.error("Don't know how to handle message type: \"%s\"", data[""], exc_info=True)
> +        logger.error(
> +            "Don't know how to handle message type: \"%s\"",
> +            data[""], exc_info=True)
>          return None
>      try:
>          returnObj = classBase()
> @@ -32,18 +46,19 @@ def constructObject(data):
>      else:
>          return returnObj
>  
> -if paths.frozen is not None:
> +
> +if paths.frozen:
>      import messagetypes.message
>      import messagetypes.vote
>  else:
> -    for mod in listdir(path.dirname(__file__)):
> +    for mod in os.listdir(os.path.dirname(__file__)):
>          if mod == "__init__.py":
>              continue
> -        splitted = path.splitext(mod)
> -        if splitted[1] != ".py":
> +        name, ext = os.path.splitext(mod)
> +        if ext != ".py":
>              continue
>          try:
> -            import_module(".{}".format(splitted[0]), "messagetypes")
> +            import_module(".{}".format(name), "messagetypes")
>          except ImportError:
>              logger.error("Error importing %s", mod, exc_info=True)
>          else:
> diff --git a/src/messagetypes/message.py b/src/messagetypes/message.py
> index f52c6b35..412b4a38 100644
> --- a/src/messagetypes/message.py
> +++ b/src/messagetypes/message.py
> @@ -3,22 +3,18 @@ from messagetypes import MsgBase
>  
>  
>  class Message(MsgBase):
> -    def __init__(self):
> -        return
> -
>      def decode(self, data):
>          # UTF-8 and variable type validator
> -        if type(data["subject"]) is str:
> +        if isinstance(data["subject"], str):
>              self.subject = unicode(data["subject"], 'utf-8', 'replace')
>          else:
>              self.subject = unicode(str(data["subject"]), 'utf-8', 'replace')
> -        if type(data["body"]) is str:
> +        if isinstance(data["body"], str):
>              self.body = unicode(data["body"], 'utf-8', 'replace')
>          else:
>              self.body = unicode(str(data["body"]), 'utf-8', 'replace')
>  
>      def encode(self, data):
> -        super(Message, self).encode()
>          try:
>              self.data["subject"] = data["subject"]
>              self.data["body"] = data["body"]
> ```

Okay. Thank you .
Please @g1itch let me know, if any further changes required.
